### ISSUE TITLE
Possible to add support for low/high night light levels?

### ISSUE THREAD

**Issue by Valdorama** [vlebourl/custom_vesync#50]
### The problem

Some air purifier (and possibly other device) models have night lights that have multiple (usually two) dimming levels. An example is the Core 200s air purifier whose night light can be set to low or high brightness.

The night light is currently implemented as a separate light entity and this works to turn it off and on. On the Core 200s, turning the light entity on turns on the night light on high brightness.

Is there any way to allow the brightness level to be controlled? Not sure how that could be done as a light entity e.g. do light entities support discrete levels rather than just on/off and dimming level?

### What version of this integration has the issue?

0.1.16

### What version of Home Assistant Core has the issue?

2022.8.7

### Diagnostics

<details><summary>Diagnostics</summary>

```json
Copy/paste diagnostics here between the starting and ending backticks.
```

</details>

### Home Assistant log

<details><summary>Logs</summary>

```py
Copy/paste any log here, between the starting and ending backticks.
```

</details>

**Comment by vlebourl** [comment]
There should be a way as light entities support levels (see the hue integration for instance). Could you add your diagnostics to the thread so I can have a look? I may ask to get access to your device at some point to test the implementation, if that's ok with you. 
Cheers

**Comment by Valdorama** [comment]
Here are the diagnostics with the night light on high brightness:

```
"data": {
    "fans": [
      {
        "device_name": "Levoit Smart True HEPA Air Purifier",
        "device_image": "https://image.vesync.com/defaultImages/deviceDefaultImages/wifibtonboarding_airpurifier_core200s_us_240.png",
        "cid": "**REDACTED**",
        "connection_status": "online",
        "connection_type": "wifi+BTOnboarding",
        "device_type": "Core200S",
        "type": "wifi-air",
        "uuid": "**REDACTED**",
        "config_module": "WiFiBTOnboarding_AirPurifier_Core200S_US",
        "mac_id": "**REDACTED**",
        "mode": "manual",
        "speed": 1,
        "extension": {
          "airQuality": -1,
          "airQualityLevel": 1,
          "mode": "manual",
          "fanSpeedLevel": "1"
        },
        "current_firm_version": null,
        "sub_device_no": null,
        "config": {
          "display": true,
          "display_forever": false
        },
        "device_status": "off",
        "enabled": false,
        "config_dict": {
          "module": "VeSyncAirBypass",
          "models": [
            "Core200S",
            "LAP-C201S-AUSR",
            "LAP-C202S-WUSR"
          ],
          "modes": [
            "sleep",
            "off"
          ],
          "features": [],
          "levels": [
            1,
            2,
            3
          ]
        },
        "features": [],
        "modes": [
          "sleep",
          "off"
        ],
        "air_quality_feature": false,
        "details": {
          "filter_life": 75,
          "mode": "manual",
          "level": 0,
          "display": false,
          "child_lock": false,
          "night_light": "on",
          "display_forever": false
        }
      }
    ]
  }
```

Diagnostics with night light on low brightness:

```
"data": {
    "fans": [
      {
        "device_name": "Levoit Smart True HEPA Air Purifier",
        "device_image": "https://image.vesync.com/defaultImages/deviceDefaultImages/wifibtonboarding_airpurifier_core200s_us_240.png",
        "cid": "**REDACTED**",
        "connection_status": "online",
        "connection_type": "wifi+BTOnboarding",
        "device_type": "Core200S",
        "type": "wifi-air",
        "uuid": "**REDACTED**",
        "config_module": "WiFiBTOnboarding_AirPurifier_Core200S_US",
        "mac_id": "**REDACTED**",
        "mode": "manual",
        "speed": 1,
        "extension": {
          "airQuality": -1,
          "airQualityLevel": 1,
          "mode": "manual",
          "fanSpeedLevel": "1"
        },
        "current_firm_version": null,
        "sub_device_no": null,
        "config": {
          "display": true,
          "display_forever": false
        },
        "device_status": "off",
        "enabled": false,
        "config_dict": {
          "module": "VeSyncAirBypass",
          "models": [
            "Core200S",
            "LAP-C201S-AUSR",
            "LAP-C202S-WUSR"
          ],
          "modes": [
            "sleep",
            "off"
          ],
          "features": [],
          "levels": [
            1,
            2,
            3
          ]
        },
        "features": [],
        "modes": [
          "sleep",
          "off"
        ],
        "air_quality_feature": false,
        "details": {
          "filter_life": 75,
          "mode": "manual",
          "level": 0,
          "display": false,
          "child_lock": false,
          "night_light": "dim",
          "display_forever": false
        }
      }
    ]
  }
```

**Comment by vlebourl** [comment]
Still active.

**Comment by Valdorama** [comment]
Is this still active..?

**Comment by vlebourl** [comment]
haven't had any time to look into it... 
I'll reopen

**Comment by Valdorama** [comment]
Thank you. No rush at all, appreciate your efforts.

**Comment by vlebourl** [comment]
looks fairly easy actually!

**Comment by vlebourl** [comment]
Could you try the branch and let me know whether that works?

**Comment by Valdorama** [comment]
Do I just replace light.py with the version from the branch and reload the integration?

**Comment by Valdorama** [comment]
I replaced light.py and I can now control the brightness of the nightlight (even though it's really only two levels):

![image](https://user-images.githubusercontent.com/34754485/191034935-2a2b296c-f4d2-4c8f-bca6-b24d7c2d64f1.png)

Thank you!

**Comment by vlebourl** [comment]
if that's usable enough, I'll publish it as is.

**Comment by Valdorama** [comment]
Yes, it's usable enough for me.

**Comment by vlebourl** [comment]
closed by #52.
